code-challenges.js

```diff
@@ -7,18 +7,59 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+//function take in a string
+//create a variable with all my vowels
+//split the word
+//use map to be able to access each value or charAt?
+//if the value = vowel change to number a=4 e=3 i=1 o=0
+//if the letter is not a vowel, return it as its itself
+//join the string together again
+//then call it
 
+const codedWords = (str) =>{
+  var str = str.split("")
+  return str.map(value => {
+    if (value === "a"){
+      return "4"
+    }
+    else if (value === "e"){
+      return "3"
+    }
+    else if (value === "i"){
+      return "1"
+    }
+    else if (value === "o"){
+      return "0"
+    }
+      else {
+        return value
+      }
+  }).join("")
+}
 
-
+console.log(codedWords(secretCodeWord1));
+console.log(codedWords(secretCodeWord2));
 
 // --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
+//function takes in array
+//filter over array to see if the values match value of "a"
+//A or a
+//returns only the words that have a in them
+//if word has value a in array, return that string
 
+const wantAs = (array) => {
+  var array = array.toString().split()
+  return array.filter(string => {
+    (string === "a" || "A")
+    return string
+  }).join("")
+}
```

Comment:
this is SOOO CLOSE. I refactored your code a little to make it work. I added comments as well to walk you through it. let me know if it helps or you have questions

```javascript
const wantAs = (array) => {
  // no need to make this a string or split it. it's already
  // an array which is "split"
  return array.filter(string => {
    // the value will only be returned if the value includes an upper
    // or lowercase "a"
    return string.includes("a") || string.includes("A")
    // no need to join, since we want to return an arary
  })
}
console.log(wantAs(arrayOfWords));
```